crates/router/src/core/payments/helpers.rs

```diff
@@ -5667,6 +5664,46 @@ where
     Ok(())
 }
 
+#[cfg(feature = "v2")]
+pub async fn override_setup_future_usage_to_on_session<F, D>(
```

Comment:
@sai-harsha-vardhan can we avoid this hanging function, also if possible can you provide relevant doc comments as to why this is being done?

Reply:
This function is not being used in payments create flow. I've created v2 version of this function because I've made setup_future_usage in payment_intent as mandatory.

Reply:
We can remove this then